coordinator/internal/authority/authority.go

```diff
@@ -130,16 +156,23 @@ func (m *Authority) WatchHistory(ctx context.Context) error {
 	}
 }
 
-// fetchState creates a fresh state from the history that's verified by the given SeedEngine.
-func (m *Authority) fetchState(se *seedengine.SeedEngine) (*State, error) {
-	latest, err := m.hist.GetLatest(&se.TransactionSigningKey().PublicKey)
+// ResetState resets the Coordinator state to the latest persisted state.
+//
+// This function is intended to be called in recovery scenarios, where a latest state exists but
+// can't be verified without the necessary secrets. The state is recovered by first loading the
+// state insecurely, then passing the corresponding manifest to the caller-provided authorization
+// function, and finally using the key material to verify the loaded state.
+//
+// The authorizeSeedSource function must check that the source of the seed (a user or a peer
+// Coordinator) are authorized to hold the secret seed, according to the manifest that's being
+// recovered to. See RFC 010 for more details on the security considerations for handling seeds.
+func (m *Authority) ResetState(oldState *State, authorizeSeedSource func(*manifest.Manifest) (*seedengine.SeedEngine, *ecdsa.PrivateKey, error)) (*State, error) {
```

Comment:
maybe use an interface instead of a function for `authorizedSeedSource`.

Reply:
Converted to interface